x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/elser_callout.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useValues } from 'kea';
+
+import {
+  EuiBadge,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPanel,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage, FormattedHTMLMessage } from '@kbn/i18n-react';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+
+export interface ELSERCallOutState {
+  dismiss: () => void;
+  dismissable: boolean;
+  show: boolean;
+}
+
+export interface ELSERCallOutProps {
+  dismissable?: boolean;
+}
+
+export const ELSER_CALL_OUT_DISMISSED_KEY = 'enterprise-search-elser-callout-dismissed';
+
+export const useELSERCallOutData = ({
+  dismissable = false,
+}: ELSERCallOutProps): ELSERCallOutState => {
+  const { supportedMLModels } = useValues(MLInferenceLogic);
+
+  const doesNotHaveELSERModel = useMemo(() => {
+    return !supportedMLModels.some((m) => m.inference_config?.text_expansion);
```

Comment:
@demjened is this the proper way to determine if the user already ELSER deployed?

Reply:
Yes, see [this](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts#L359). Note we call this feature "text expansion" instead of "ELSER" internally.